src/plan/immix/global.rs

```diff
@@ -139,6 +140,7 @@ impl<VM: VMBinding> Immix<VM> {
                 unlog_object_when_traced: false,
                 #[cfg(feature = "vo_bit")]
                 mixed_age: false,
+                never_move_objects: cfg!(feature = "immix_non_moving"),
```

Comment:
This line `never_move_objects: cfg!(feature = "immix_non_moving")` is repeated three times for Immix, GenImmix and StickyImmix.  But the semantics of the Cargo feature "immix_non_moving" is disallowing *any* ImmixSpace instance to move objects.  Repeating `cfg!(feature = "immix_non_moving")` imposes a danger that in the future, as we add more spaces or plans, we may miss one or two of this, and accidentally assign `true` to `never_move_objects` when the "immix_non_moving" feature is enabled.

I suggest we allow plans to set `never_move_objects` according to the requirements in the "normal" scenario (i.e. the main ImmixSpace is movable, but the dedicated non-moving space is non-movable), and let `ImmixSpace::new` override its value according to `cfg!(feature = "immix_non_moving")`.  In this way, we guarantee `never_move_objects == false` if  the "immix_non_moving" feature is enabled.

Suggestion:
```suggestion
                never_move_objects: true,
```

But override it in `ImmixSpace::new`.